### PR TITLE
Wallet handle type safety

### PR DESCRIPTION
**Pull request by AxelNennker**
Re-introduce type-safety for WalletHandle in libindy
Introduce type-saftety for WalletHandle in wrappers/rust and libindy/tests

----
Not unrelated note: That the integration tests use wrappers/rust and use private code from libindy feels wrong to me. 
https://github.com/hyperledger/indy-sdk/blob/master/libindy/Cargo.toml#L95
https://github.com/hyperledger/indy-sdk/blob/master/libindy/tests/utils/mod.rs#L7

From this sins follows the next, like this weirdness:
```rust
extern crate indyrs as indy;
extern crate indyrs as api;
```
https://github.com/hyperledger/indy-sdk/blob/master/libindy/tests/anoncreds.rs#L17
Madness.

### CODE REVIEW

**Comment by AxelNennker** [comment]
cli and vcx need some massaging

File: cli/src/commands/mod.rs [hunk 1]
```diff
@@ -185,51 +185,63 @@ pub fn set_active_did(ctx: &CommandContext, did: Option<String>) {
...
+        Some((WalletHandle(handle), name))
     } else {
         None
     }
 }
 
-pub fn get_opened_wallet_handle(ctx: &CommandContext) -> Option<i32> {
-    ctx.get_int_value("OPENED_WALLET_HANDLE")
+pub fn get_opened_wallet_handle(ctx: &CommandContext) -> Option<WalletHandle> {
+    match ctx.get_int_value("OPENED_WALLET_HANDLE") {
```
Comment (jovfer) [inline cli/src/commands/mod.rs]:
```suggestion
    match ctx.get_int_value("OPENED_WALLET_HANDLE").map(|val| WalletHandle(val))
```

File: cli/src/commands/mod.rs [hunk 2]
```diff
@@ -185,51 +185,63 @@ pub fn set_active_did(ctx: &CommandContext, did: Option<String>) {
...
-    ctx.set_string_value("OPENED_WALLET_NAME", value.as_ref().map(|value| value.1.to_owned()));
-    ctx.set_sub_prompt(2, value.map(|value| format!("wallet({})", value.1)));
+pub fn set_opened_wallet(ctx: &CommandContext, value: Option<(WalletHandle, String)>) {
+    match value {
+        Some((wallet_handle, wallet_name)) => {
+            ctx.set_int_value("OPENED_WALLET_HANDLE", Some(wallet_handle.0));
+            ctx.set_string_value("OPENED_WALLET_NAME", Some(wallet_name.to_owned()));
+            ctx.set_sub_prompt(2, Some(wallet_name));
+        },
+        _ => {
```
Comment (jovfer) [inline cli/src/commands/mod.rs]:
```suggestion
        None => {
```

File: cli/src/commands/mod.rs (hunk 1 above)
Comment (jovfer) [reply cli/src/commands/mod.rs]: and next few lines can be removed

**Comment by AxelNennker** [comment]
Not really sure why this build fails. I think the fail is not caused by this PR

**Comment by AxelNennker** [comment]
A [commit](https://github.com/hyperledger/indy-sdk/pull/1877/commits/f0c71e18bb28a390a7d5fcb78f73447e279a28eb) in this PR improves the error message that leads to this build failure.

The error message for this build failure is "Could not connect Error(Hyper(Error(Connect, Os { code: 99, kind: AddrNotAvailable, message: "Cannot assign requested address" })), "http://localhost:8080/agency/msg")"

This has nothing to do with this PR, I think. Probably more of a build system error.
Could somebody please have a look into this?!

**Comment by AxelNennker** [comment]
The build error is unrelated to this PR
```
thread 'main' panicked at 'Unable to locate libzmq:

`"pkg-config" "--libs" "--cflags" "libzmq >= 4.1"` did not exit successfully: exit code: 1

--- stderr

Package libzmq was not found in the pkg-config search path.

Perhaps you should add the directory containing `libzmq.pc'

to the PKG_CONFIG_PATH environment variable

No package 'libzmq' found
```

**Comment by AxelNennker** [comment]
CI still can't find libzmq. I think this is unrelated to this PR.

File: wrappers/rust/Cargo.toml [hunk 3]
```diff
@@ -18,6 +18,7 @@ path = "src/lib.rs"
 crate-type = ["rlib"]
 
 [dependencies]
+indy-api-types = { path = "../../libindy/indy-api-types"}
```
Comment (jovfer) [inline wrappers/rust/Cargo.toml]: TODO: update CI/CD to publish `indy-api-types` crate before the wrapper itself

File: wrappers/rust/indy-sys/src/lib.rs [hunk 4]
```diff
@@ -1,5 +1,10 @@
 extern crate libc;
 
+#[macro_use] extern crate serde_derive;
```
Comment (jovfer) [inline wrappers/rust/indy-sys/src/lib.rs]: Is it really used now in `indy-sys`?

**Pull request merged**